# HG changeset patch
# User Pantxo Diribarne <pantxo.diribarne@gmail.com>
# Date 1427442938 -3600
#      Fri Mar 27 08:55:38 2015 +0100
# Node ID 3f461c2b990e3c945036264fb9a58bb879806704
# Parent  4840fc43f0b5481a92840e7965026076651aa957
Don't reset toolkit when the name hasn't changed

* graphics.in.h (figure::properties::set___graphics_toolkit__ ()): do not
  reset the value of the __graphics_toolkit__ if it doesn't change.
  Avoids the figure to be finalized/re-initialized.

diff --git a/libinterp/corefcn/graphics.in.h b/libinterp/corefcn/graphics.in.h
--- a/libinterp/corefcn/graphics.in.h
+++ b/libinterp/corefcn/graphics.in.h
@@ -3396,18 +3396,21 @@ public:
               std::string nm = val.string_value ();
               graphics_toolkit b = gtk_manager::find_toolkit (nm);
               if (b.get_name () != nm)
                 {
                   error ("set___graphics_toolkit__: invalid graphics toolkit");
                 }
               else
                 {
-                  set_toolkit (b);
-                  mark_modified ();
+                  if (nm != get___graphics_toolkit__ ())
+                    {
+                      set_toolkit (b);
+                      mark_modified ();
+                    }
                 }
             }
           else
             error ("set___graphics_toolkit__ must be a string");
         }
     }
 
     void adopt (const graphics_handle& h);
